Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -35,6 +35,25 @@ public override void Initialize()
             // We want to process lung reagents before we inhale new reagents.
             UpdatesAfter.Add(typeof(MetabolizerSystem));
             SubscribeLocalEvent<RespiratorComponent, ApplyMetabolicMultiplierEvent>(OnApplyMetabolicMultiplier);
+
+            SubscribeLocalEvent<RespiratorImmunityComponent, ComponentInit>(OnPressureImmuneInit);
+            SubscribeLocalEvent<RespiratorImmunityComponent, ComponentRemove>(OnPressureImmuneRemove);
+        }
+
+        private void OnPressureImmuneInit(EntityUid uid, RespiratorImmunityComponent pressureImmunity, ComponentInit args)
+        {
+            if (TryComp<RespiratorComponent>(uid, out var respirator))
+            {
+                respirator.HasImmunity = true;
+            }
+        }
+
+        private void OnPressureImmuneRemove(EntityUid uid, RespiratorImmunityComponent pressureImmunity, ComponentRemove args)
+        {
+            if (TryComp<RespiratorComponent>(uid, out var respirator))
+            {
+                respirator.HasImmunity = false;
+            }
```

Comment:
This should all be replaced with firing a cancellable event in TakeSuffocationDamage.

Reply:
no since it would still gasp and show the alert even if damage isnt dealt
mob composition is a thing now so you can just not have it need to respire